### PR TITLE
Updated Q

### PR DESCRIPTION
**Pull request by dionboles-asym**
changes in Q 
- Q  updated  to a Python folder module
- Q is easier to update it as a dotenv file for updates and docker file PORT changes 
- Q has Total time execution on run function now so it will show how long a query will take in Milliseconds
- Q understands with the single_operator_parser function you can do single line Q statements
- Q updated the unique_term function to add limit #39

### CODE REVIEW

File: cdapython/Q.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+from cdapython.constantVariables import CDA_API_URL, table_version, default_table
+from cdapython.Result import get_query_result
+from cdapython.functions import Quoted, Unquoted, col
+from cda_client.api.meta_api import MetaApi
+from cdapython.decorators import measure
+
+
+class Q:
+    """
+    Q lang is Language used to send query to the cda servi
```
Comment (nemanjab17) [inline cdapython/Q.py:14]: service*

File: cdapython/Qparser.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+class Eq(Token):
+    lbp = 10  # Precedence
+
+    def led(self, left: str, context):
+        """Compute the value of this token when between two expressions."""
+        # Fetch the expression to the right, stoping at the next boundary
+        # of same precedence
+        right_side = context.expression(self.lbp)
+        return Q(left.strip()+" = "+right_side.strip())
```
Comment (nemanjab17) [inline cdapython/Qparser.py:23]: maybe add spaces between class definitions

File: tests/test_integration.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-
+import pytest
 
 def test_basic_integration():
     cols = columns()
     assert "race" in cols
 
 
 def test_unique_terms():
+    # pytest.set_trace()
```
Comment (nemanjab17) [inline tests/test_integration.py:11]: what is the purprose of this?

**Review by nemanjab17 — APPROVED** [review]

**Comment by elijahlowe** [comment]
single_operator_parser (renamed query) does not seem to parse if operator is not "=". see below

```
q = query('ResearchSubject.associated_project = "TCGA-OV" AND ResearchSubject.Diagnosis.age_at_diagnosis = 21550')

r = q.run()

print(r)
```

```
Getting results from database

Total execution time: 1231 ms

QueryID: f361c5b0-845b-4245-ac40-4b60f785fc5d
Query: SELECT v3.* FROM gdc-bq-sample.cda_mvp.v3 AS v3, UNNEST(ResearchSubject) AS _ResearchSubject, UNNEST(_ResearchSubject.Diagnosis) AS _Diagnosis WHERE ((_ResearchSubject.associated_project = 'TCGA-OV') AND (_Diagnosis.age_at_diagnosis = 21550))
Offset: 0
Count: 1
Total Row Count: 1
More pages: False
```

While similar query only switching the "=" to "<" brakes the search:

```
q = query('ResearchSubject.associated_project = "TCGA-OV" AND ResearchSubject.Diagnosis.age_at_diagnosis < 21550')

r = q.run()

print(r)
```

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-67-b6337f878c84> in <module>
----> 1 q = query('ResearchSubject.associated_project = "TCGA-OV" AND ResearchSubject.Diagnosis.age_at_diagnosis < 21550')
      2 
      3 r = q.run()
      4 
      5 print(r)

~/Desktop/test/cda-python/cdapython/utility.py in query(text)
     11 
     12 def query(text: str) -> Q:
---> 13     return parser(text)
     14 
     15 

~/Desktop/test/cda-python/cdapython/Qparser.py in parser(text)
    171 
    172 def parser(text):
--> 173     return lexer.parse(text)

~/opt/miniconda3/envs/cda/lib/python3.9/site-packages/tdparser/lexer.py in parse(self, text)
    163         tokens = self.lex(text)
    164         parser = Parser(tokens)
--> 165         return parser.parse()

~/opt/miniconda3/envs/cda/lib/python3.9/site-packages/tdparser/topdown.py in parse(self)
    212     def parse(self):
    213         """Parse the flow of tokens, and return their evaluation."""
--> 214         expr = self.expression()
    215         if not isinstance(self.current_token, EndToken):
    216             raise InvalidTokenError("Unconsumed trailing tokens.")

~/opt/miniconda3/envs/cda/lib/python3.9/site-packages/tdparser/topdown.py in expression(self, rbp)
    206             # than to the right of an expression.
    207             prev_token = self.consume()
--> 208             left = prev_token.led(left, context=self)
    209 
    210         return left

~/Desktop/test/cda-python/cdapython/Qparser.py in led(self, left, context)
    125         # of same precedence
    126         right_side = context.expression(self.lbp)
--> 127         return left.And(right_side)
    128 
    129 

~/Desktop/test/cda-python/cdapython/Q.py in And(self, right)
    151 
    152     def And(self, right: "Q") -> "Q":
--> 153         return Q(self.query, "AND", right.query)
    154 
    155     def Or(self, right: "Q") -> "Q":

AttributeError: 'str' object has no attribute 'query'
```

**Review by elijahlowe — APPROVED** [review]

**Pull request merged**